src/index.js

```diff
@@ -220,18 +240,18 @@ class Dropzone extends React.Component {
     this.node = ref;
   }
 
-  fileAccepted(file) {
+  fileAccepted(file, props = this.props) {
     // Firefox versions prior to 53 return a bogus MIME type for every file drag, so dragovers with
     // that MIME type will always be accepted
-    return file.type === 'application/x-moz-file' || accepts(file, this.props.accept);
+    return file.type === 'application/x-moz-file' || accepts(file, props.accept);
   }
 
   fileMatchSize(file) {
     return file.size <= this.props.maxSize && file.size >= this.props.minSize;
   }
 
-  allFilesAccepted(files) {
-    return files.every(this.fileAccepted);
+  allFilesAccepted(files, props = this.props) {
```

Comment:
Same here, let's make the `accept` a required second arguement to make those functions pure.